### ISSUE TITLE
[Bug Report] [Component] [dropdown, popconfirm] el-dropdown in slot causes el-popconfirm, which uses slot data, position to be reset back to 0,0 after trigger el-popconfirm

### ISSUE THREAD

**Issue by Greesea** [element-plus/element-plus#7037]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.1.8`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.75 Safari/537.36`
- Build Tool: `Vue CLI`

### Related Component

- `el-dropdown`
- `el-popconfirm`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8Q29tcCAjc2xvdD1cInNjb3BlXCI+XG4gICAgPGVsLWRyb3Bkb3duPlxuICAgICAgPGRpdj5kcm9wZG93biB0cmlnZ2VyPC9kaXY+XG4gICAgICA8dGVtcGxhdGUgI2Ryb3Bkb3duPlxuICAgICAgICA8ZWwtZHJvcGRvd24tbWVudT5cbiAgICAgICAgICA8ZWwtZHJvcGRvd24taXRlbT5cbiAgICAgICAgICAgIDxlbC1wb3Bjb25maXJtIDp0aXRsZT1cImAke3Njb3BlfWBcIiAjcmVmZXJlbmNlPlxuICAgICAgICAgICAgICA8ZGl2PnBvcGNvbmZpcm0gdHJpZ2dlcjwvZGl2PlxuICAgICAgICAgICAgPC9lbC1wb3Bjb25maXJtPlxuICAgICAgICAgIDwvZWwtZHJvcGRvd24taXRlbT5cbiAgICAgICAgPC9lbC1kcm9wZG93bi1tZW51PlxuICAgICAgPC90ZW1wbGF0ZT5cbiAgICA8L2VsLWRyb3Bkb3duPlxuICA8L0NvbXA+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IHNldHVwPlxuICBpbXBvcnQgQ29tcCBmcm9tIFwiLi9Db21wLnZ1ZVwiO1xuPC9zY3JpcHQ+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XCJpbXBvcnRzXCI6e319IiwiQ29tcC52dWUiOiI8dGVtcGxhdGU+XG4gIDxkaXY+XG4gICAgPHNsb3QgbmFtZT1cInNsb3RcIiB2LWJpbmQ9XCJkYXRhXCIvPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgc2V0dXA+XG4gIGltcG9ydCB7cmVmfSBmcm9tIFwidnVlXCI7XG4gIFxuICBjb25zdCBkYXRhID0gcmVmKFt7djogMX0sIHt2OiAyfV0pO1xuPC9zY3JpcHQ+XG4ifQ==)

### Steps to reproduce

1. hover to activate el-dropdown
2. click el-dropdown-item to activate el-popconfirm
3. el-popconfirm opened at correct position and suddenly teleport to 0,0

if el-dropdown with hide-on-click=true then you need to move the cursor out of el-dropdown-item after step 2

[zh-CN]
1. 鼠标悬浮在文本上激活 el-dropdown
2. 点击 el-dropdown-item 激活 el-popconfirm
3. 此时 el-popconfirm 在正确位置打开后因 el-dropdown 关闭立刻移动至 0,0

如果设置 hide-on-click=true 给 el-dropdown 则需要在第二步之后将鼠标移出 el-dropdown-item 的范围也能重现相同问题

### What is Expected?

el-popconfirm keep it position at activate

[zh-CN]
el-popconfirm 保持在打开时的位置

### What is actually happening?

el-popconfirm suddenly teleport to 0,0

[zh-CN]
el-popconfirm 在激活时瞬间被移动到 0,0 的位置

### Additional comments

the problem is that everything is fine when el-popconfirm is not using slot data. i guess it's el-dropdown or Vue Slot mechanism caused this issue.

[zh-CN]
在 el-popconfirm 上不使用 slot 的数据时一切都正常，我怀疑是 el-dropdown 或者是 Vue 的插槽机制导致的问题

**Comment by Greesea** [comment]
looks like this issue has been fix at 2.2.0
这个问题在2.2.0疑似已被修正

**Comment by frynt** [comment]
on the exemple i have *dropdown-menu* => *dropdown-item* => *el-popconfirm*

**Comment by Greesea** [comment]
> Still reproduced. See the [playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxkaXYgc3R5bGU9XCJwb3NpdGlvbjphYnNvbHV0ZTsgcmlnaHQ6IDE2MHB4OyB0b3A6IDE2MHB4XCI+XG4gIDxlbC1kcm9wZG93biB0cmlnZ2VyPVwiY2xpY2tcIj5cbiAgICA8YiBzdHlsZT1cImZvbnQtc2l6ZTogMzBweFwiPkNsaWNrIG1lPC9iPlxuICAgIDx0ZW1wbGF0ZSAjZHJvcGRvd24+XG4gICAgICA8ZWwtZHJvcGRvd24tbWVudT5cbiAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0+XG4gICAgICAgICAgPGVsLXRvb2x0aXAgY29udGVudD1cIkltIGEgdG9vbHRpcFwiPlxuICAgICAgICAgIFx0PGRpdj5cbiAgICAgICAgICAgICAgPGVsLXBvcGNvbmZpcm0gdGl0bGU9XCJBcmUgeW91IHN1cmVcIj5cbiAgICAgICAgICAgICAgPHRlbXBsYXRlICNyZWZlcmVuY2U+XG4gICAgICAgICAgICAgICAgQ2xpY2sgbWUgYW5kIHNlZSBwb3Bjb25maXJtIHRlbGVwb3J0ZWQgdG8gMCwwXG4gICAgICAgICAgICAgICAgPC90ZW1wbGF0ZT5cbiAgICAgICAgICAgICAgPC9lbC1wb3Bjb25maXJtPlxuICAgICAgICAgIFx0PC9kaXY+XG4gICAgICAgICAgPC9lbC10b29sdGlwPlxuICAgICAgICA8L2VsLWRyb3Bkb3duLWl0ZW0+XG4gICAgICA8L2VsLWRyb3Bkb3duLW1lbnU+XG4gICAgPC90ZW1wbGF0ZT5cbiAgPC9lbC1kcm9wZG93bj5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319). Be careful it may require sometimes 20 click to reproduce the bug ...
> 
> on the exemple i have `<el-tootlip>` :
> 
> `<dropdown-menu>` => `<dropdown-item>` => `<el-tooltip>` => `<el-popconfirm>`
> 
> ![image](https://user-images.githubusercontent.com/11840996/186137158-84e5a085-9b73-4dc7-9e1d-3d8be5241b94.png)

my demo works on latest Vue with Element Plus 2.2.0 but not works on latest Element Plus.
this issue has been ignored for entire 2 month and no dev care about it. currently i'm not using Element Plus anymore so maybe you can open another issue for this bug.
